File: platform/polyglot/extract-imports-text-based-notation/py/extract_imports_text_based_notation/specifier.py
import enum


class SpecifierKind(enum.StrEnum):
    DEFAULT = "default"
    NAMED = "named"
    NAMESPACE = "namespace"
    SIDE_EFFECT = "side-effect"
    EXPORT_DEFAULT = "export-default"
    EXPORT_NAMED = "export-named"
    EXPORT_ALL = "export-all"
    EXPORT_NAMESPACE = "export-namespace"


def format_named(name: str, alias: str | None = None) -> str:
    if alias and alias != name:
        return f"named: {name} as {alias}"
    return f"named: {name}"


def format_namespace(name: str, alias: str | None = None) -> str:
    if alias and alias != name:
        return f"namespace: {name} as {alias}"
    return f"namespace: {name}"


def format_export_named(name: str, alias: str | None = None) -> str:
    if alias and alias != name:
        return f"export-named: {name} as {alias}"
    return f"export-named: {name}"

File: platform/polyglot/extract-imports-text-based-notation/mjs/src/specifier.ts
// ── Import specifier formatters ──────────────────────────────────────────────

export function formatDefault(localName: string): string {
  return `default: ${localName}`;
}

export function formatNamed(importedName: string, localName: string): string {
  if (importedName === localName) {
    return `named: ${importedName}`;
  }
  return `named: ${importedName} as ${localName}`;
}

export function formatNamespace(localName: string): string {
  return `namespace: ${localName}`;
}

// ── Export specifier formatters ─────────────────────────────────────────────

export function formatExportDefault(name: string): string {
  return `export-default: ${name}`;
}

export function formatExportNamed(name: string, alias?: string): string {
  if (alias !== undefined && alias !== name) {
    return `export-named: ${name} as ${alias}`;
  }
  return `export-named: ${name}`;
}

export function formatExportAll(): string {
  return 'export-all';
}

export function formatExportNamespace(name: string): string {
  return `export-namespace: ${name}`;
}
